sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/npm.yaml

```diff
@@ -0,0 +1,45 @@
+provider:
+  metadata:
+    name: NPM
+    category: Package repository
+    message: Make sure this NPM API token gets revoked, changed, and removed from the code.
+  detection:
+    pre:
+      include:
+        content:
+          - npm_
+    post:
+      patternNot:
+        - "(?i)(?:s|ex)ample|foo|bar|test|abcd|redacted"
+
+  rules:
+    - id: npm-access-token
+      rspecKey: S6758
+      metadata:
+        name: NPM access tokens should not be disclosed
+      detection:
+        matching:
+          pattern: "(npm_[0-9A-Za-z]{36})\\b"
```

Comment:
Based on this search [sourcegraph](https://sourcegraph.com/search?q=context:global+/%28npm_%5B0-9A-Za-z%5D%7B36%7D%29%5C%5Cb/&patternType=standard&case=yes&sm=1&groupBy=repo) it appears there are a few false positives like `npm_XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX` that could be fixed with a statistical post filter.

Suggested threshold: 4.26